deployments/yunikorn-application/sample-yunikorn-app.yaml

```diff
@@ -0,0 +1,24 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: "yunikorn.apache.org/v1alpha1"
+kind: Application
+metadata:
+  name: example
```

Comment:
we need to be careful here. will this field be used as our unique `applicationID`?
I guess this name can same across namespaces, correct? so most likely it cannot avoid dup.
And in most cases, users may not want to input appID, we should be able to generate one for them. How can we achieve that? Maybe use the object UID?
I think we need to think more about these problems before moving forward.

Reply:
Adding a name is mandatory, and yes it will be unique only in the namespace. We can introduce a new field as ApplicationID and if it is not filled, we can generate it. I would export in some common place and use the same generator, used by the admission controller. What do you think?

Reply:
An another idea is to build the applicationID from the namespace + name. When querying the application in case of a status update we need to provide the namespace and the name, but from the core side we will get only the applicationID. If we don't have the name then I would need to list all the apps and find the one with a specific applicationID, but if I can parse the name from the applicationID, it can be queried for a given namespace and name.

Reply:
hi @kingamarton 

According to https://kubernetes.io/docs/concepts/overview/working-with-objects/names/. Object UID should be unique across the cluster. Maybe we can use that as the `applicationID`. That means the user only needs to specify a namespace-unique name for each app-CRD. Also, metadata has `generateName` field, so if the user doesn't specify any name, K8s will generate one with that as the prefix. What do you think about this?